src/Jmikola/AutoLogin/Http/Firewall/AutoLoginListener.php

```diff
@@ -106,6 +122,14 @@ public function handle(GetResponseEvent $event)
             if ($authenticatedToken = $this->authenticationManager->authenticate($token)) {
                 $this->securityContext->setToken($authenticatedToken);
 
+                //return response to redirect after successful login
+                $response = $this->successHandler->onAuthenticationSuccess($request, $authenticatedToken);
```

Comment:
Is the AuthenticationSuccessHandlerInterface mainly used for creating redirect responses? I don't recall using these before, but I'd be curious to see other listeners (either in Symfony or outside) that use this service if you have prior art to link to.

It's been a while, but I believe this component purposefully avoids setting a response object or otherwise getting in the way of the request.